TABTestKit/Classes/BasePage.swift

```diff
@@ -10,34 +10,24 @@
 
 import XCTest
 
-/// All pages should inherit from this class
-protocol PageProtocol {
-  var trait: XCUIElement? { get }
+public protocol UITestPage: class {
+  var trait: XCUIElement { get }
 }
 
-open class BasePage: GlobalFunctions, PageProtocol {
+public extension UITestPage {
+  /// Use primarily to assert that you're on the correct screen - all screens should try to have a trait element.
+  /// If the trait is static it can be decalred in the init() function of a page, if you want it to be dynamic you can pass it an element to use as a trait
+  func await(maxDuration duration: Double = 5) {
+    GlobalFunctions().waitForElementToAppear(trait, maxDuration: duration)
+  }
+}
+
+open class BasePage {
   
-  var trait: XCUIElement?
+  public init() { }
   
-  public override init() {}
-
   /// Native back button control
   public func tapBackButton() {
-    let app = XCUIApplication()
-    
-    app.navigationBars.children(matching: .button).matching(identifier: "Back").element(boundBy: 0).tap()
+    App.shared.navigationBars.children(matching: .button).matching(identifier: "Back").element(boundBy: 0).tap()
```

Comment:
We probably don't want that in the pod as not every app will have this button.  Could we get rid of base page all together as this is the only function in it?

Reply:
Surely most apps have this? Isn't it a required Apple thing?

Reply:
@KITSTABSamBerry do we run the risk of this library/pod not needing to exist if we strip everything out?